tfhe-zk-pok/src/proofs/pke.rs

```diff
@@ -1805,6 +1820,48 @@ mod tests {
         }
     }
 
+    /// Test encryption of a message where the delta used for encryption is not the one used for
+    /// proof/verify
+    #[test]
+    fn test_bad_delta() {
+        let PkeTestParameters {
+            d,
+            k,
+            B,
+            q,
+            t,
+            msbs_zero_padding_bit_count,
+        } = PKEV1_TEST_PARAMS;
+
+        let effective_cleartext_t = t >> msbs_zero_padding_bit_count;
+
+        let rng = &mut StdRng::seed_from_u64(0);
+
+        let testcase = PkeTestcase::gen(rng, PKEV1_TEST_PARAMS);
+        let mut testcase_bad_delta = testcase.clone();
+
+        // Make sure that the messages lower bit is set so the change of delta has an impact on the
+        // validity of the ct
+        testcase_bad_delta.m = (0..k)
+            .map(|_| (rng.gen::<u64>() % effective_cleartext_t) as i64 & 1)
+            .collect::<Vec<_>>();
+
+        // Encrypt using wrong delta
+        let ct = testcase.encrypt(PKEV1_TEST_PARAMS_BAD_DELTA);
```

Comment:
update good params on the fly to be bad, avoids having to manage another constant and avoids to have to keep it in sync